### PR TITLE
Add multirun_with_transition macro.

### PR DESCRIPTION
**Pull request by oxidase**
Added a macro to have multirun rules with configurations other than "target".

### CODE REVIEW

File: multirun.bzl [hunk 1]
```diff
@@ -105,6 +105,15 @@ multirun = rule(
             cfg = "exec",
             executable = True,
         ),
-    },
-    executable = True,
-)
+    }
+
+    if allowlist:
+        attrs["_allowlist_function_transition"] =  attr.label(default = allowlist)
```
Comment (keith) [inline multirun.bzl:111]: do we need to pass this or can we just always set this even if it's not used? (maybe there's a downside to that though?)

**Review by keith — APPROVED** [review]
nice, can you add a small test target exercising this?

File: multirun.bzl (hunk 1 above)
Comment (oxidase) [reply multirun.bzl:111]: * for configurations with a state transition and _without_ `_allowlist_function_transition ` attribute Bazel complains
```
ERROR: .../rules_multirun/multirun.bzl:113:16: Use of Starlark transition without allowlist attribute '_allowlist_function_transition'. See Starlark transitions documentation for details and usage: //deploy:aws.bzl NORMAL
```
which is a TODO to be removed https://github.com/bazelbuild/bazel/blob/e2189245/src/main/java/com/google/devtools/build/lib/analysis/starlark/StarlarkRuleClassFunctions.java#L923-L935

* for "exec" and "target" configuration and _with_ _allowlist_function_transition ` attribute Bazel complains
```
ERROR: .../rules_multirun/multirun.bzl:113:16: Unused function-based split transition allowlist: //deploy:aws.bzl NORMAL
```
in the else branch at https://github.com/bazelbuild/bazel/blob/e2189245/src/main/java/com/google/devtools/build/lib/analysis/starlark/StarlarkRuleClassFunctions.java#L937-L944

>  just always set this even if it's not used
it is not possible to set the attribute unconditionally.

**Comment by oxidase** [comment]
> nice, can you add a small test target exercising this?

added `//tests::multirun_with_transition` to `//tests:test`

**Comment by keith** [comment]
looks good! but looks like you need to add a dep on platforms for bzlmod

**Comment by oxidase** [comment]
> need to add a dep on platforms for bzlmod
I removed references to `//platforms` to avoid adding `bazel_dep(name = "platforms", version = "0.0.6")` just for testing

**Review by keith — APPROVED** [review]
nice!

**Pull request merged**